src/main/java/com/hermesworld/ais/galapagos/devcerts/impl/DeveloperCertificateServiceImpl.java

```diff
@@ -119,6 +123,25 @@ public void init(KafkaCluster cluster) {
         return Optional.of(metadata);
     }
 
+    @Override
+    public CompletableFuture<Integer> clearExpiredDeveloperCertificatesOnAllClusters() {
+        CompletableFuture<Void> result = FutureUtil.noop();
+        AtomicInteger totalClearedDevCerts = new AtomicInteger();
+
+        for (KafkaCluster cluster : kafkaClusters.getEnvironments()) {
+            List<DevCertificateMetadata> expiredDevCerts = getRepository(cluster).getObjects().stream()
+                    .filter(devCert -> devCert.getExpiryDate().isBefore(timeService.getTimestamp().toInstant()))
+                    .collect(Collectors.toList());
+            for (DevCertificateMetadata cert : expiredDevCerts) {
+                result = result.thenCompose(future -> aclUpdater.removeAcls(cluster, Collections.singleton(cert)))
```

Comment:
Well, removeAcls allows to pass a SET of certificates. Why not passing all `expiredDevCerts` at once? (They have to be collected as Set, not as List, above)